lib/eu_central_bank.rb

```diff
@@ -99,54 +91,58 @@ def set_rate(from, to, rate, date = nil)
   end
 
   def rates
-    store.each_rate.each_with_object({}) do |(from,to,rate,date),hash|
-      key = [from, to].join(SERIALIZER_SEPARATOR)
-      key = [key, date.to_s].join(SERIALIZER_DATE_SEPARATOR) if date
-      hash[key] = rate
+    store.each_rate.each_with_object({}) do |(from, to, rate, date), hash|
+      hash[date] ||= []
+      hash[date] << { from: from, to: to, rate: rate }
+    end
+  end
+
+  def save_rates(file_path, url = ECB_RATES_URL)
+    raise Errors::InvalidFilePath unless file_path
+
+    File.open(file_path, 'w') do |file|
+      io = open(url)
+      io.each_line { |line| file.puts line }
     end
   end
 
-  def export_rates(format, file = nil, opts = {})
-    raise Money::Bank::UnknownRateFormat unless
-      RATE_FORMATS.include? format
+  def export_rates(format)
+    raise Money::Bank::UnknownRateFormat unless RATE_FORMATS.include? format
 
     store.transaction true do
-      s = case format
+      case format
       when :json
         JSON.dump(rates)
       when :ruby
         Marshal.dump(rates)
       when :yaml
         YAML.dump(rates)
       end
-
-      unless file.nil?
-        File.open(file, "w") {|f| f.write(s) }
-      end
-
-      s
     end
   end
 
-  def import_rates(format, s, opts = {})
-    raise Money::Bank::UnknownRateFormat unless
-      RATE_FORMATS.include? format
+  def import_rates(format, content)
+    raise Money::Bank::UnknownRateFormat unless RATE_FORMATS.include? format
 
     store.transaction true do
       data = case format
-       when :json
-         JSON.load(s)
-       when :ruby
-         Marshal.load(s)
-       when :yaml
-         YAML.load(s)
-       end
-
-      data.each do |key, rate|
-        from, to = key.split(SERIALIZER_SEPARATOR)
-        to, date = to.split(SERIALIZER_DATE_SEPARATOR)
-
-        store.add_rate from, to, BigDecimal(rate), date
+             when :json
+               JSON.parse(content, symbolize_names: true)
+             when :ruby
+               Marshal.load(content)
+             when :yaml
+               YAML.load(content)
+      end
```

Comment:
I wonder if it should be `import_rate`'s responsibility to know about all those formats.